src/tmpfiles/tmpfiles.c

```diff
@@ -2839,6 +2840,39 @@ static int create_item(Context *c, Item *i) {
         return 0;
 }
 
+static int purge_item_instance(Context *c, Item *i, const char *instance, CreationMode creation) {
+        int r;
+
+        /* FIXME: we probably should use dir_cleanup() here instead of rm_rf() so that 'x' is honoured. */
+        log_debug("rm -rf \"%s\"", instance);
+        r = rm_rf(instance, REMOVE_ROOT|REMOVE_SUBVOLUME|REMOVE_PHYSICAL);
+        if (r < 0 && r != -ENOENT)
+                return log_error_errno(r, "rm_rf(%s): %m", instance);
+
+        return 0;
+}
+
+static int purge_item(Context *c, Item *i) {
+
+        assert(i);
+
+        log_debug("Running purge owned action for entry %c %s", (char) i->type, i->path);
+
+        if (IN_SET(i->type,
+                   COPY_FILES,
+                   TRUNCATE_FILE,
+                   CREATE_FILE,
+                   WRITE_FILE,
+                   EMPTY_DIRECTORY,
+                   CREATE_SYMLINK,
+                   CREATE_FIFO,
+                   CREATE_DIRECTORY,
+                   TRUNCATE_DIRECTORY))
```

Comment:
i'd prefer if this list was moved somewhere to the top next to needs_glob() and takes_ownership() in a new helper func, so that the categorization of actions is unified at a single place